api-authorization/server/lib/authorization-middleware.ts

```diff
@@ -0,0 +1,42 @@
+/* eslint-disable @typescript-eslint/no-unused-vars -- Remove me */
+import { NextFunction, Request, Response } from 'express';
+import jwt from 'jsonwebtoken';
+import { ClientError } from './client-error.js';
+
+export function authMiddleware(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  /* your code here */
+  try {
+    const authHeader = req.get('Authorization');
+    if (!authHeader) throw new ClientError(401, 'Authentication required');
+    const token = authHeader.split('Bearer ')[1];
+    if (!token) throw new ClientError(401, 'Authentication required');
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET as string);
+    (req as any).user = payload;
+    next();
+  } catch (error) {
+    if (error instanceof jwt.JsonWebTokenError) {
```

Comment:
This is handled for you in error-middleware.ts so you don't need it here. Actually, in middleware like this you don't even want the try/catch since you want the exception to bubble up the stack until Express takes care of it. So you can remove the entire try/catch.